src/wbi/experiment.py

```diff
@@ -125,160 +124,59 @@ def generate_cameraframedata_files(self, output_folder=None):
-
-        if self.frames_sync.latency_shift:
-            frame_count_daq += self.frames_sync.latency_shift
-            logger.info(f"Applying latency shift {self.frames_sync.latency_shift}")
-
-        d_frame_count = np.diff(frame_count)
-
-        for i in np.arange(len(frame_count) - 1):
-            if d_frame_count[i] == 0 and d_frame_count[i - 1] == 2:
-                frame_count[i] -= 1
-
-        volume_direction = np.empty(len(frame_count))
-        volume_direction[:] = np.nan
-        piezo_position = np.full(frame_count.shape, np.nan, dtype=float)
-        for i, element in enumerate(frame_count):
-            try:
-                index_in_daq = (
-                    np.where(frame_count_daq == element)[0][0]
-                    if len(np.where(frame_count_daq == element)[0]) > 0
-                    else None
-                )
-                piezo_position[i] = self.frames_sync.sync["piezo_position"][
-                    index_in_daq
-                ]
-                volume_direction[i] = (
-                    self.frames_sync.sync["piezo_direction"][index_in_daq] + 1
-                ) // 2
-            except KeyError:
-                pass
-
-        nans, non_zeros = np.isnan(piezo_position), lambda z: z.nonzero()[0]
-        piezo_position[nans] = np.interp(
-            non_zeros(nans), non_zeros(~nans), piezo_position[~nans]
-        )
-        volume_direction[nans] = np.interp(
-            non_zeros(nans), non_zeros(~nans), volume_direction[~nans]
-        ).astype(np.int8)
-
-        smoothing_window_size = config.flash_finder.piezo_smoothing_window_size
-        smoothing_window = np.ones(smoothing_window_size) / smoothing_window_size
-        piezo_sm = np.convolve(piezo_position, smoothing_window, mode="same")
-
-        volume_direction[:-1] = np.sign(np.diff(piezo_sm))
-        volume_direction[-1] = volume_direction[-2]
-        volume_direction = (volume_direction + 1) // 2
-        volume_index = np.cumsum(np.absolute(np.diff(volume_direction)))
-
-        return volume_index
-
     def flash_finder(self, output_folder=None, chunk_size=4000, max_frames=None):
         output_folder = output_folder or self.folder_path
 
         dat = self.dat
-        timing = self.timing
-
-        brightness = np.zeros(dat.n_frames)
-        stdev = np.zeros(dat.n_frames)
+        n_frames = dat.n_frames if max_frames is None else min(max_frames, dat.n_frames)
+        brightness = np.zeros(n_frames)
+        stdev = np.zeros(n_frames)
 
-        for index, chunk in enumerate(dat.chunks(chunk_size=chunk_size)):
-            chunk = np.transpose(chunk, (2, 1, 0)).reshape(-1, dat.cols * dat.rows * 2)
+        for index, chunk in enumerate(
+            dat.chunks(chunk_size=chunk_size, max_frames=n_frames)
+        ):
             brightness[index * chunk_size : (index + 1) * chunk_size] = np.average(
-                chunk, axis=1
+                chunk, axis=(0, 1)
```

Comment:
It's perfectly fine to take avg/std etc. on multiple axes, so there's no need to transpose, reshape etc. The result you get this way will be in the same shape as what you would without this simplfication.